paywall/src/hooks/components/Web3Service.js

```diff
@@ -0,0 +1,16 @@
+import PropTypes from 'prop-types'
+import React, { createContext } from 'react'
+
+import web3Service from '../../services/web3Service'
+
+export const Web3ServiceContext = createContext()
```

Comment:
```suggestion
export const Web3Context = createContext()
```

Reply:
renamed to `Web3ServiceProvider`. The suffix `Provider` should be used for all context providers, and the prefix should describe what it is passing down through context. So, for example, the existing `ConfigProvider` component is a great example of this naming convention. This component passes down the `web3Service` through its context. PTAL